src/main/java/frc/team2412/robot/commands/autonomous/AutonomousCommand.java

```diff
@@ -58,7 +120,49 @@ public Command getAutonomousCommand(Trajectory exampleTrajectory) {
                 thetaController,
                 drivebaseSubsystem::updateModules,
                 drivebaseSubsystem);
+        SwerveControllerCommand swerveControllerCommandFour = new SwerveControllerCommand(
+                trajectoryFour,
+                drivebaseSubsystem::getPoseAsPoseMeters, // Functional interface to feed supplier
+                AutoConstants.driveKinematics,
+
+                // Position controllers
+                new PIDController(AutoConstants.PX_CONTROLLER, 0, 0),
+                new PIDController(AutoConstants.PY_CONTROLLER, 0, 0),
+                thetaController,
+                drivebaseSubsystem::updateModules,
+                drivebaseSubsystem);
+
+        addCommands(
+                // new ParallelCommandGroup(
+                // new ScheduleCommand(new ShooterTargetCommand(shooterSubsystem, localizer)),
+                // new WaitCommand(1)),
+                // new ParallelDeadlineGroup(new WaitCommand(1), new IndexShootCommand(indexSubsystem)),
+                // new IntakeSetExtendCommand(intakeSubsystem),
+                new ParallelCommandGroup(
+                        // paths
+                        new SequentialCommandGroup(swerveControllerCommandOne, swerveControllerCommandThree,
+                                swerveControllerCommandFour)),
+                // actions
+                // STEPS FOR COMMAND
+                // 1. drive to ball 2, intake,
+                // 2. drive to ball 3, intake, shoot
+                // 3. drive to ball 4, intake
+                // 4. wait 2ish seconds, intake,
+                // 5. drive, shoot
+
+                new SequentialCommandGroup(
```

Comment:
Instead of creating a new command group, you should combine both these commands and the swerve controller commands into one command group. That way you won't have to rely on timing to make sure the commands run at the right times.

Reply:
Very good idea, I implemented it